dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ExecutorServiceImpl.java

```diff
@@ -75,6 +76,23 @@ public final class ExecutorServiceImpl implements WSExecutorService, ThreadQuies
      */
     ThreadPoolController threadPoolController = null;
 
+    /**
+     * Receive notification when server start completes
+     */
+
+    @Reference(policy = ReferencePolicy.DYNAMIC, cardinality = ReferenceCardinality.OPTIONAL)
+    protected synchronized void setServerStarted(ServerStarted serverStarted) {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+            Tr.event(tc, ": server start complete.");
+        }
+        threadPoolController.startupCompleted();
```

Comment:
What happens when a large number of applications are configured and take longer than the 30 seconds to start?  That defaults to registering the `ServerStarted` service before all the applications have really started.  If that has a very large set of applications to start will it then grow the pool to be "too big" and result in large contention again?

Reply:
Understood